build/tools/Automation/DotNet/DotNetAutomator.cs

```diff
@@ -69,6 +63,17 @@ public async Task RunAsync()
             }
         }
 
+        public async Task InitializeFieldsAsync()
```

Comment:
nit: let's make this a `private` method since it's only accessed within the scope of this class

Reply:
Just want to double check here. I made this a public method so the private fields could be initialized in test. Making this private will make the newly added unit test fail since mock can't access private method. Is it ok to remove the added unit test once I make this method private or should we keep the public method and the unit test here?

Reply:
OK in that case I think we should scrap this new method and move the instantiation of these variables to be in the constructor of this class -- that way they're still accessible in the tests since we're calling the constructor directly

Reply:
I think if we put this in the constructor, it'll bypass any mock http requests, which then causes the unit test to fail as well. Is there any other ways to get around this?

Reply:
Discussed offline -- we should be good with the current implementation based on how the class and its methods are being tested.